artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/AddressImpl.java

```diff
@@ -37,7 +36,7 @@
 
    private final boolean containsWildCard;
 
-   private List<Address> linkedAddresses = null;
+   private final List<Address> linkedAddresses = new ArrayList<>();
```

Comment:
Whats the impact here for those not using wildcards

Reply:
there is an impact, it is reverted to lazy init via https://github.com/apache/activemq-artemis/pull/3343 where a set works better for deletions.